### ISSUE TITLE
Insert by default and `:` keyword 

### ISSUE THREAD

**Issue by fabswt** [jsumners/alfred-emoji#63]
Hi there,

I've been trying to do these things:

- **Use colon (`:`) as the keyword**, without a space (for consistency with, say, Notion, GitHub, and to speed things up) So as to look for, say, `:clown`.
  - It works... to a limit: results include (distracting) non-emoji results, e.g.: `:a` will return both "Activity monitor" and other programs, along a couple of emojis. Any option to not fall back to non-emoji results?
- **Make insert ("Paste {query} into frontmost application") the default behavior** to speed things up.
  - I was successful making the change (just unticked the 'Command' modifier key in the workflow on the one brand, to tick it and set modifier subtext "Copy "{query}" to clipboard" on the other brand.) Problems are: the default text still remains "Copy ... to clipboard", and it copies it to clipboard anyway, which is not desirable.)
  - I tried editing the plist (and then restarting Alfred) to edit the phrasings, but it's been without impact. What did I miss?

Thanks for the help!

**Comment by jsumners** [comment]
The `:<thing>` syntax is used by those editors as a way for their parser to pick up that the user is trying to type an emoji character. Alfred works on keywords, as you noted, and those require a separator, e.g. this suggestion would become `: clown`. You are welcome to set your own keyword for the workflow via the workflow preferences; not that you'd likely have to adjust the keyword after every workflow update.

The current default behavior follows the standard Alfred behavior. Making the change to "insert on enter" would make this workflow counter intuitive to Alfred users.

**Comment by fabswt** [comment]
I'm not sure I follow for the editor thing... When I type `:<thing>` in an editor, I'm right in the editor. When I type `:<thing>` inside of the Alfred bar, the editor does not know about it. My question was how come some of the results come from outside the real of emojis (there's visibly a form of fallback mechanism in the search, disabling it would get rid of the irrelevant results.) This depends on the search, see:
  - Works fine with query `:smile`:
    <img width="567" alt="Screen Shot 2021-11-22 at 2 37 36 PM" src="https://user-images.githubusercontent.com/3042488/142871417-9719f9f2-f06e-42d4-bd26-6ad655e325c6.png">
  - Getting irrelevant results with query `:a`:
    <img width="572" alt="Screen Shot 2021-11-22 at 2 37 26 PM" src="https://user-images.githubusercontent.com/3042488/142871457-f99de4a7-0be6-4a5e-913e-8433754ef956.png">

I understand most users may want to keep using copy as the default behavior. But how would I go about making "insert on enter" the default behavior?

**Comment by jsumners** [comment]
Did you change the keyword of the action and uncheck "with space"?

**Comment by fabswt** [comment]
Yes.

**Comment by jsumners** [comment]
Then it is clear doing so will not work.

**Comment by fabswt** [comment]
Do you know why it's working with `:smile` but providing non-Emoji results with `:a`?

**Comment by jsumners** [comment]
No, I don't. You'd have to investigate how Alfred interprets keywords without a space in combination with the workflow's script. I do not have time to do this.

**Comment by fabswt** [comment]
Sure, no worries, just asking.

**Comment by rmm5t** [comment]
@fabswt Personally, I use a keyword snippet in Alfred to trigger this workflow. All my snippets start with a period (<kbd>.</kbd>), so my trigger is <kbd>.</kbd><kbd>:</kbd>

> But how would I go about making "insert on enter" the default behavior? (if only for me, or as an option.)

 That's a feature you can enable. Just set the `snippetapp` environment variable (to any value) in the workflow configuration to change the behavior to always paste the emoji character regardless of how the workflow is instantiated.

More information can be found on the forum thread about this starting here:
https://www.alfredforum.com/topic/11126-alfred-emoji-search-emojis-by-name-or-keyword/?do=findComment&comment=58339

Here's a screenshot of it too:
![image](https://user-images.githubusercontent.com/740/143066360-74eefa99-9a36-4991-8e1e-cd7c15f8501d.png)

**Comment by rmm5t** [comment]
> Do you know why it's working with :smile but providing non-Emoji results with :a?

@fabswt Agreed with @jsumners here. I think this is a side effect of Alfred, not necessarily this emoji workflow. That might be a good question for the general Alfred forum. 

@fabswt Do you think this issue needs to remain open, or are you okay with closing this out now?

**Comment by jsumners** [comment]
> That's a feature you can enable. Just set the snippetapp environment variable (to any value) in the workflow configuration to change the behavior to always paste the emoji character regardless of how the workflow is instantiated.

@rmm5t I can't recall, and I'm on my phone right now, did we add this to the docs?

**Comment by rmm5t** [comment]
@jsumners  It's a feature I use daily. It doesn't look like it's in our README. Only documented on that forum thread.

**Comment by jsumners** [comment]
Maybe we should document it.

**Comment by rmm5t** [comment]
> Maybe we should document it.

I’m out running Thanksgiving errands, but I can tackle that later if no one else submits a PR in the meantime. 😉

**Comment by jsumners** [comment]
Certainly no rush. 🦃🦃🦃

**Comment by qubist** [comment]
I think macOS also uses `:` and `/` specially in file paths, which could be related to why searching `:a` is bringing up things with paths starting with `:A`.
